docs/source/search/plugin.rst

```diff
@@ -0,0 +1,48 @@
+API Reference
+*************
+
+The ``IndicoSearchProvider`` interface allows an abstract service to integrate with Indico's search module.
```

Comment:
you can probably use something like this to link to the actual doc of that class:

    :class:`IndicoSearchProvider`